packages/dom-adapters/src/InlineToolAdapter/index.ts

```diff
@@ -0,0 +1,144 @@
+import type {
+  EditorJSModel,
+  InlineFragment,
+  InlineToolData,
+  InlineToolName,
+  ModelEvents,
+  TextRange
+} from '@editorjs/model';
+import {
+  EventType,
+  TextFormattedEvent,
+  TextUnformattedEvent
+} from '@editorjs/model';
+import type { CaretAdapter } from '../CaretAdapter/index.js';
+import { type IntersectType, FormattingAction } from '@editorjs/model';
+
+export interface InlineTool {
+  name: InlineToolName;
+  intersectType: IntersectType,
+  create(data?: InlineToolData): HTMLElement;
+  getAction(index: TextRange, fragments: InlineFragment[], intersectType: IntersectType, data?: InlineToolData): { action: FormattingAction; range: TextRange };
+}
+
+/**
+ * Class handles on format model events and renders inline tools
+ * Applies format to the model
+ */
+export class InlineToolAdapter {
```

Comment:
If it would be a single adapter for all of the tools, let's name it `InlineToolsAdapter`